src/internal/pfsdb/commits.go

```diff
@@ -0,0 +1,956 @@
+	"github.com/pachyderm/pachyderm/v2/src/internal/pachsql"
+	"github.com/pachyderm/pachyderm/v2/src/internal/stream"
+	"github.com/pachyderm/pachyderm/v2/src/pfs"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+const (
+	// CommitsChannelName is used to watch events for the commits table.
+	CommitsChannelName = "pfs_commits"
+	createCommit       = `
+		WITH repo_row_id AS (SELECT id from pfs.repos WHERE name=:repo_name AND type=:repo_type AND project_id=(SELECT id from core.projects WHERE name=:proj_name))
+		INSERT INTO pfs.commits 
+    	(commit_id, 
+    	 commit_set_id, 
+    	 repo_id, 
+    	 branch_id, 
+    	 description, 
+    	 origin, 
+    	 start_time, 
+    	 finishing_time, 
+    	 finished_time, 
+    	 compacting_time_s, 
+    	 validating_time_s, 
+    	 size, 
+    	 error) 
+		VALUES 
+		(:commit_id, 
+		 :commit_set_id, 
+		 (SELECT id from repo_row_id), 
+		(SELECT id from pfs.branches WHERE name=:branch_name AND repo_id=(SELECT id from repo_row_id)), 
+		:description, 
+		:origin, 
+		:start_time,
+		:finishing_time,
+		:finished_time,
+		:compacting_time_s,
+		:validating_time_s,
+		:size,
+		:error)
+		RETURNING int_id;`
+	updateCommit = `
+		WITH repo_row_id AS (SELECT id from pfs.repos WHERE name=:repo_name AND type=:repo_type AND project_id=(SELECT id from core.projects WHERE name=:proj_name))
+		UPDATE pfs.commits SET 
+			commit_id=:commit_id, 
+			commit_set_id=:commit_set_id,
+		    repo_id=(SELECT id from repo_row_id), 
+		    branch_id=(SELECT id from pfs.branches WHERE name=:branch_name AND repo_id=(SELECT id from repo_row_id)), 
+			description=:description, 
+			origin=:origin, 
+			start_time=:start_time, 
+			finishing_time=:finishing_time, 
+			finished_time=:finished_time, 
+		    compacting_time_s=:compacting_time_s, 
+			validating_time_s=:validating_time_s, 
+			size=:size, 
+			error=:error 
+		WHERE int_id=:int_id;`
+	getCommit = `
+		SELECT 
+    		commit.int_id, 
+    		commit.commit_id, 
+    		commit.repo_id, 
+    		commit.branch_id, 
+    		commit.origin, 
+    		commit.description, 
+    		commit.start_time, 
+    		commit.finishing_time, 
+    		commit.finished_time, 
+    		commit.compacting_time_s, 
+    		commit.validating_time_s,
+    		commit.error, 
+    		commit.size, 
+    		repo.name AS repo_name, 
+    		repo.type AS repo_type, 
+    		project.name AS proj_name, 
+    		branch.name as branch_name
+		FROM pfs.commits commit
+		JOIN pfs.repos repo ON commit.repo_id = repo.id
+		JOIN core.projects project ON repo.project_id = project.id
+		LEFT JOIN pfs.branches branch ON commit.branch_id = branch.id`
+	getParentCommit = getCommit + `
+		JOIN pfs.commit_ancestry ancestry ON ancestry.from_id = commit.int_id`
+	getChildCommit = getCommit + `
+		JOIN pfs.commit_ancestry ancestry ON ancestry.to_id = commit.int_id`
+)
+
+// ErrCommitNotFound is returned by GetCommit() when a commit is not found in postgres.
+type ErrCommitNotFound struct {
```

Comment:
consider: do you think it's apt time to move all custom errors in this package to a separate `errors.go` file?

Reply:
I think it actually makes more sense to group errors near their related type. But I don't have a strong preference. I'll defer to the team's wider style on this one.